app/assets/javascripts/deploys.js

```diff
@@ -13,13 +16,22 @@ $(function () {
       $form = $("#new_deploy"),
       $submit = $form.find('input[type=submit]');
 
-  // load changeset when switching to it
-  $("#deploy-tabs a[data-type=github]").click(function (e) {
+  // load changeset when switching to it (for deploy#show)
+  $("#deploy-tabs a[data-toggle=tab][href^='#']").click(function (e) {
       e.preventDefault();
-      var tab = $(this);
-      tab.tab("show");
 
+      var $tab = $(this);
+      var $placeholder = $container.find('.changeset-placeholder');
+      $tab.tab("show");
+
+      // abort if already loaded (either static or via previous changeset load)
+      if ($container.find($tab.attr('href')).length) { return; }
+
+      // prevent double-loading
       if (!changesetLoaded) {
```

Comment:
this also prevents double-loading again ?

Reply:
yep, but also prevents requesting a second time while the ajax request is in flight (ie. a double click on the tab)